From 8d6dcd880899cc7bcb74197918fea8dbda88d7bd Mon Sep 17 00:00:00 2001
From: rpm-build <rpm-build>
Date: Tue, 10 Jun 2025 15:33:02 +0200
Subject: [PATCH 2/3] add: rhel 10.0, rhel 10.1 (prerelease)

Update rhel-10-unknown to match 10.2 and later.

62f8adce rhel: Add RHEL 10.1 (prerelease)
2a374acd rhel10: set release date, add test data

Related: RHEL-87787
Signed-off-by: rpm-build <rpm-build>
---
 os/redhat.com/rhel-10-unknown.xml |  28 ++---
 os/redhat.com/rhel-10.0.xml       |  10 +-
 os/redhat.com/rhel-10.1.xml       | 168 ++++++++++++++++++++++++++++++
 3 files changed, 187 insertions(+), 19 deletions(-)
 create mode 100644 os/redhat.com/rhel-10.1.xml

diff --git a/os/redhat.com/rhel-10-unknown.xml b/os/redhat.com/rhel-10-unknown.xml
index ee3519e..d464e8e 100644
--- a/os/redhat.com/rhel-10-unknown.xml
+++ b/os/redhat.com/rhel-10-unknown.xml
@@ -15,13 +15,13 @@
     <name xml:lang="fr">Red Hat Enterprise Linux 10 inconnu</name>
     <version>10-unknown</version>
     <vendor>Red Hat, Inc</vendor>
-    <vendor xml:lang="zh_Hans">Red Hat, Inc</vendor>
+    <vendor xml:lang="zh-Hans">Red Hat, Inc</vendor>
     <vendor xml:lang="ka">Red Hat, Inc</vendor>
     <vendor xml:lang="ko">Red Hat, Inc</vendor>
     <vendor xml:lang="fi">Red Hat, Inc</vendor>
     <vendor xml:lang="uk">Red Hat, Inc</vendor>
     <vendor xml:lang="tr">Red Hat, Inc</vendor>
-    <vendor xml:lang="pt_BR">Red Hat, Inc</vendor>
+    <vendor xml:lang="pt-BR">Red Hat, Inc</vendor>
     <vendor xml:lang="pl">Red Hat, Inc</vendor>
     <vendor xml:lang="ja">Red Hat, Inc</vendor>
     <vendor xml:lang="it">Red Hat, Inc</vendor>
@@ -33,59 +33,59 @@
     <family>linux</family>
     <distro>rhel</distro>
     <codename>Coughlan</codename>
-    <upgrades id="http://redhat.com/rhel/10.0"/>
-    <derives-from id="http://redhat.com/rhel/10.0"/>
+    <upgrades id="http://redhat.com/rhel/10.1"/>
+    <derives-from id="http://redhat.com/rhel/10.1"/>
     <media arch="x86_64">
       <iso>
-        <volume-id>RHEL-10-([1-9]|[1-9][0-9])(-[0-9]|[1-9][0-9])?.*x86_64</volume-id>
+        <volume-id>RHEL-10-([2-9]|[1-9][0-9])(-[0-9]|[1-9][0-9])?.*x86_64</volume-id>
       </iso>
-      <kernel>isolinux/vmlinuz</kernel>
-      <initrd>isolinux/initrd.img</initrd>
+      <kernel>images/pxeboot/vmlinuz</kernel>
+      <initrd>images/pxeboot/initrd.img</initrd>
     </media>
     <media arch="aarch64">
       <iso>
-        <volume-id>RHEL-10-([1-9]|[1-9][0-9])(-[0-9]|[1-9][0-9])?.*aarch64$</volume-id>
+        <volume-id>RHEL-10-([2-9]|[1-9][0-9])(-[0-9]|[1-9][0-9])?.*aarch64$</volume-id>
       </iso>
       <kernel>images/pxeboot/vmlinuz</kernel>
       <initrd>images/pxeboot/initrd.img</initrd>
     </media>
     <media arch="ppc64le">
       <iso>
-        <volume-id>RHEL[_-]10[_-]([1-9]|[1-9][0-9])([_-]([0-9]|[1-9][0-9]))?.*ppc64le$</volume-id>
+        <volume-id>RHEL[_-]10[_-]([2-9]|[1-9][0-9])([_-]([0-9]|[1-9][0-9]))?.*ppc64le$</volume-id>
       </iso>
       <kernel>ppc/ppc64/vmlinuz</kernel>
       <initrd>ppc/ppc64/initrd.img</initrd>
     </media>
     <media arch="s390x">
       <iso>
-        <volume-id>RHEL-10-([1-9]|[1-9][0-9])(-[0-9]|[1-9][0-9])?.*s390x$</volume-id>
+        <volume-id>RHEL-10-([2-9]|[1-9][0-9])(-[0-9]|[1-9][0-9])?.*s390x$</volume-id>
       </iso>
     </media>
     <tree arch="aarch64">
       <treeinfo>
         <family>Red Hat Enterprise Linux</family>
-        <version>^10.([1-9]|[1-9][0-9])(-[0-9]|[1-9][0-9])?$</version>
+        <version>^10.([2-9]|[1-9][0-9])(-[0-9]|[1-9][0-9])?$</version>
         <arch>aarch64</arch>
       </treeinfo>
     </tree>
     <tree arch="ppc64le">
       <treeinfo>
         <family>Red Hat Enterprise Linux</family>
-        <version>^10.([1-9]|[1-9][0-9])(.([0-9]|[1-9][0-9]))?$</version>
+        <version>^10.([2-9]|[1-9][0-9])(.([0-9]|[1-9][0-9]))?$</version>
         <arch>ppc64le</arch>
       </treeinfo>
     </tree>
     <tree arch="s390x">
       <treeinfo>
         <family>Red Hat Enterprise Linux</family>
-        <version>^10.([1-9]|[1-9][0-9])(.([0-9]|[1-9][0-9]))?$</version>
+        <version>^10.([2-9]|[1-9][0-9])(.([0-9]|[1-9][0-9]))?$</version>
         <arch>s390x</arch>
       </treeinfo>
     </tree>
     <tree arch="x86_64">
       <treeinfo>
         <family>Red Hat Enterprise Linux</family>
-        <version>^10.([1-9]|[1-9][0-9])(.([0-9]|[1-9][0-9]))?$</version>
+        <version>^10.([2-9]|[1-9][0-9])(.([0-9]|[1-9][0-9]))?$</version>
         <arch>x86_64</arch>
       </treeinfo>
     </tree>
diff --git a/os/redhat.com/rhel-10.0.xml b/os/redhat.com/rhel-10.0.xml
index 9f11c48..cdc2b57 100644
--- a/os/redhat.com/rhel-10.0.xml
+++ b/os/redhat.com/rhel-10.0.xml
@@ -15,13 +15,13 @@
     <name xml:lang="fr">Red Hat Enterprise Linux 10.0</name>
     <version>10.0</version>
     <vendor>Red Hat, Inc</vendor>
-    <vendor xml:lang="zh_Hans">Red Hat, Inc</vendor>
+    <vendor xml:lang="zh-Hans">Red Hat, Inc</vendor>
     <vendor xml:lang="ka">Red Hat, Inc</vendor>
     <vendor xml:lang="ko">Red Hat, Inc</vendor>
     <vendor xml:lang="fi">Red Hat, Inc</vendor>
     <vendor xml:lang="uk">Red Hat, Inc</vendor>
     <vendor xml:lang="tr">Red Hat, Inc</vendor>
-    <vendor xml:lang="pt_BR">Red Hat, Inc</vendor>
+    <vendor xml:lang="pt-BR">Red Hat, Inc</vendor>
     <vendor xml:lang="pl">Red Hat, Inc</vendor>
     <vendor xml:lang="ja">Red Hat, Inc</vendor>
     <vendor xml:lang="it">Red Hat, Inc</vendor>
@@ -35,13 +35,13 @@
     <codename>Coughlan</codename>
     <upgrades id="http://redhat.com/rhel/9.5"/>
     <derives-from id="http://redhat.com/rhel/9.5"/>
-    <release-status>prerelease</release-status>
+    <release-date>2025-05-20</release-date>
     <media arch="x86_64">
       <iso>
         <volume-id>RHEL-10-0(-[0-9]|[1-9][0-9])?.*x86_64</volume-id>
       </iso>
-      <kernel>isolinux/vmlinuz</kernel>
-      <initrd>isolinux/initrd.img</initrd>
+      <kernel>images/pxeboot/vmlinuz</kernel>
+      <initrd>images/pxeboot/initrd.img</initrd>
     </media>
     <media arch="aarch64">
       <iso>
diff --git a/os/redhat.com/rhel-10.1.xml b/os/redhat.com/rhel-10.1.xml
new file mode 100644
index 0000000..16aa649
--- /dev/null
+++ b/os/redhat.com/rhel-10.1.xml
@@ -0,0 +1,168 @@
+<?xml version="1.0"?>
+<libosinfo version="0.0.1">
+  <!-- Licensed under the GNU General Public License version 2 or later.
+     See http://www.gnu.org/licenses/ for a copy of the license text -->
+  <os id="http://redhat.com/rhel/10.1">
+    <short-id>rhel10.1</short-id>
+    <name>Red Hat Enterprise Linux 10.1</name>
+    <version>10.1</version>
+    <vendor>Red Hat, Inc</vendor>
+    <vendor xml:lang="zh-Hans">Red Hat, Inc</vendor>
+    <vendor xml:lang="ka">Red Hat, Inc</vendor>
+    <vendor xml:lang="ko">Red Hat, Inc</vendor>
+    <vendor xml:lang="fi">Red Hat, Inc</vendor>
+    <vendor xml:lang="uk">Red Hat, Inc</vendor>
+    <vendor xml:lang="tr">Red Hat, Inc</vendor>
+    <vendor xml:lang="pt-BR">Red Hat, Inc</vendor>
+    <vendor xml:lang="pl">Red Hat, Inc</vendor>
+    <vendor xml:lang="ja">Red Hat, Inc</vendor>
+    <vendor xml:lang="it">Red Hat, Inc</vendor>
+    <vendor xml:lang="id">Red Hat, Inc</vendor>
+    <vendor xml:lang="fr">Red Hat, Inc</vendor>
+    <vendor xml:lang="es">Red Hat, Inc</vendor>
+    <vendor xml:lang="de">Red Hat, Inc</vendor>
+    <vendor xml:lang="ca">Red Hat, Inc</vendor>
+    <family>linux</family>
+    <distro>rhel</distro>
+    <codename>Coughlan</codename>
+    <upgrades id="http://redhat.com/rhel/10.0"/>
+    <derives-from id="http://redhat.com/rhel/10.0"/>
+    <release-status>prerelease</release-status>
+    <media arch="x86_64">
+      <iso>
+        <volume-id>RHEL-10-1(-[0-9]|[1-9][0-9])?.*x86_64</volume-id>
+      </iso>
+      <kernel>images/pxeboot/vmlinuz</kernel>
+      <initrd>images/pxeboot/initrd.img</initrd>
+    </media>
+    <media arch="aarch64">
+      <iso>
+        <volume-id>RHEL-10-1(-[0-9]|[1-9][0-9])?.*aarch64$</volume-id>
+      </iso>
+      <kernel>images/pxeboot/vmlinuz</kernel>
+      <initrd>images/pxeboot/initrd.img</initrd>
+    </media>
+    <media arch="ppc64le">
+      <iso>
+        <volume-id>RHEL[_-]10[_-]1([_-]([0-9]|[1-9][0-9]))?.*ppc64le$</volume-id>
+      </iso>
+      <kernel>ppc/ppc64/vmlinuz</kernel>
+      <initrd>ppc/ppc64/initrd.img</initrd>
+    </media>
+    <media arch="s390x">
+      <iso>
+        <volume-id>RHEL-10-1(-[0-9]|[1-9][0-9])?.*s390x$</volume-id>
+      </iso>
+    </media>
+    <tree arch="aarch64">
+      <treeinfo>
+        <family>Red Hat Enterprise Linux</family>
+        <version>^10.1(-[0-9]|[1-9][0-9])?$</version>
+        <arch>aarch64</arch>
+      </treeinfo>
+    </tree>
+    <tree arch="ppc64le">
+      <treeinfo>
+        <family>Red Hat Enterprise Linux</family>
+        <version>^10.1(.([0-9]|[1-9][0-9]))?$</version>
+        <arch>ppc64le</arch>
+      </treeinfo>
+    </tree>
+    <tree arch="s390x">
+      <treeinfo>
+        <family>Red Hat Enterprise Linux</family>
+        <version>^10.1(.([0-9]|[1-9][0-9]))?$</version>
+        <arch>s390x</arch>
+      </treeinfo>
+    </tree>
+    <tree arch="x86_64">
+      <treeinfo>
+        <family>Red Hat Enterprise Linux</family>
+        <version>^10.1(.([0-9]|[1-9][0-9]))?$</version>
+        <arch>x86_64</arch>
+      </treeinfo>
+    </tree>
+    <resources arch="ppc64le">
+      <minimum>
+        <n-cpus>1</n-cpus>
+        <cpu>1000000000</cpu>
+        <ram>2147483648</ram>
+        <storage>10737418240</storage>
+      </minimum>
+      <recommended>
+        <cpu>4000000000</cpu>
+        <ram>2147483648</ram>
+        <storage>21474836480</storage>
+      </recommended>
+      <maximum>
+        <n-cpus>768</n-cpus>
+        <ram>35184372088832</ram>
+      </maximum>
+    </resources>
+    <resources arch="x86_64">
+      <minimum>
+        <cpu>1000000000</cpu>
+        <ram>1610612736</ram>
+        <storage>10737418240</storage>
+      </minimum>
+      <recommended>
+        <cpu>1000000000</cpu>
+        <ram>1610612736</ram>
+        <storage>21474836480</storage>
+      </recommended>
+      <maximum>
+        <n-cpus>768</n-cpus>
+        <ram>26388279066624</ram>
+      </maximum>
+      <network-install>
+        <ram>3221225472</ram>
+      </network-install>
+    </resources>
+    <resources arch="s390x">
+      <minimum>
+        <ram>1073741824</ram>
+        <storage>10737418240</storage>
+      </minimum>
+      <recommended>
+        <storage>21474836480</storage>
+      </recommended>
+      <maximum>
+        <n-cpus>340</n-cpus>
+        <ram>17592186044416</ram>
+      </maximum>
+      <network-install>
+        <ram>2147483648</ram>
+      </network-install>
+    </resources>
+    <resources arch="aarch64">
+      <minimum>
+        <ram>2147483648</ram>
+        <storage>10737418240</storage>
+      </minimum>
+      <recommended>
+        <storage>21474836480</storage>
+      </recommended>
+      <maximum>
+        <n-cpus>256</n-cpus>
+        <ram>1649267441664</ram>
+      </maximum>
+    </resources>
+    <resources arch="all">
+      <minimum>
+        <n-cpus>1</n-cpus>
+        <cpu>1000000000</cpu>
+        <ram>1610612736</ram>
+        <storage>10737418240</storage>
+      </minimum>
+      <recommended>
+        <cpu>1000000000</cpu>
+        <ram>1610612736</ram>
+        <storage>21474836480</storage>
+      </recommended>
+    </resources>
+    <installer>
+      <script id="http://redhat.com/rhel/kickstart/jeos"/>
+      <script id="http://redhat.com/rhel/kickstart/desktop"/>
+    </installer>
+  </os>
+</libosinfo>
-- 
2.49.0

